back/app/controllers/web_api/v1/users_controller.rb

```diff
@@ -110,6 +110,21 @@ def by_invite
     show
   end
 
+  # To validate an email without creating a user and return whether to show terms if they are have no password
+  def check
+    @user = User.find_by email: params[:email]
+    skip_authorization
+    if !User::EMAIL_REGEX.match?(params[:email])
+      render json: { errors: { email: [{ error: 'invalid', value: params[:email] }] } }, status: :unprocessable_entity
+    elsif @user && !@user&.no_password?
+      render json: { errors: { email: [{ error: 'taken', value: params[:email] }] } }, status: :unprocessable_entity
+    elsif @user&.registration_completed_at.present?
+      render json: { action: 'continue' }
+    else
+      render json: { action: 'show_terms' }
+    end
```

Comment:
I wonder if we could just send the status of the user/email, instead of a prescriptive action. For instance, we could return something like 'new_user,' and then the front-end would decide what's best to do (show the terms and conditions in this case).